flink-tests/src/test/java/org/apache/flink/test/checkpointing/AutoRescalingITCase.java

```diff
@@ -427,7 +427,8 @@ public void testCheckpointRescalingWithKeyedAndNonPartitionedState() throws Exce
 
             restClusterClient.updateJobResourceRequirements(jobID, builder.build()).join();
 
-            waitForRunningTasks(restClusterClient, jobID, parallelism2);
+            // Source is parallelism, the flatMapper & Sink is parallelism2
+            waitForRunningTasks(restClusterClient, jobID, parallelism + parallelism2);
```

Comment:
Can't we also reduce the cooldown phase to make the test complete faster similarly to what you did in https://github.com/1996fanrui/flink/commit/ffd713e24d37db2c103e4cd4361d0cd916d0d2f6? I tried it locally but it didn't help, though. :thinking:

Reply:
I checked that the shortening the cooldown phase works in PR #24246 . Not sure what I meant initially. :innocent: